### PR TITLE
Read ARTIFACTORY_* build args before running Yarn/NPM

### PR DESCRIPTION
**Pull request by henrik242**
This allows for an `.npmrc` with variables:
```
_auth="${ARTIFACTORY_NPM_SECRET}"
always-auth=true
email="${ARTIFACTORY_USER}"
registry="${ARTIFACTORY_CONTEXT}/api/npm/npm-virtual"
```
Use with e.g.
```
docker build --build-arg ARTIFACTORY_USER=me@example.com ...
```

### CODE REVIEW

File: Dockerfile.onbuild [hunk 1]
```diff
@@ -1,5 +1,9 @@
 FROM containers.schibsted.io/finntech/node:0.0.0
 
+ONBUILD ARG ARTIFACTORY_USER
+ONBUILD ARG ARTIFACTORY_NPM_SECRET
```
Comment (ivarconr) [inline Dockerfile.onbuild]: Generally building a secret in to the image sounds like a anti-pattern. 

Why would you need to access artifactory from a docker image?

Comment (henrik242) [reply Dockerfile.onbuild]: We run yarn inside the image as part of the build process

Comment (henrik242) [reply Dockerfile.onbuild]: (Which is just part of the `scripts/install-dependencies.sh` script provided by the onbuild image)

Comment (henrik242) [reply Dockerfile.onbuild]: Also, the credentials will not be a part of the image, it's just used to build the image

Comment (trygve-lie) [reply Dockerfile.onbuild]: What is the problem this solves?

Comment (henrik242) [reply Dockerfile.onbuild]: So, this image copies `.npmrc` (amongst other files) to the image, and runs `yarn install` (or `npm install`) inside it as part of the build process.

Our `.npmrc` looks like this: 
```
_auth="${ARTIFACTORY_NPM_SECRET}"
always-auth=true
email="${ARTIFACTORY_USER}"
registry="${ARTIFACTORY_CONTEXT}/api/npm/npm-virtual"
```
This makes yarn/npm access repositories that need authentication (e.g. `artifacts.schibsted.io`) _without_ having to store authentication within the image.

Comment (trygve-lie) [reply Dockerfile.onbuild]: So we are storing authentication information in the images today?

Comment (henrik242) [reply Dockerfile.onbuild]: No, we are usually using npm.finntech.no, which is a proxy against artifacts.schibsted.io that doesn't require a password.  However, this has been notoriously unstable for us for months, especially in Travis.  All devs in our team has set the ARTIFACTORY_ variables locally, and we use a copy of

**Pull request merged**